packager/mpd/base/xml/xml_node.cc

```diff
@@ -56,14 +64,15 @@ bool IsTimelineConstantDuration(const std::list<SegmentInfo>& segment_infos,
     return false;
 
   const SegmentInfo& first_segment = segment_infos.front();
-  if (first_segment.start_time != first_segment.duration * (start_number - 1))
+  if (!ApproximiatelyEqual(first_segment.start_time,
```

Comment:
It may be difficult to be approximately equal with 5% error for live streaming.

I suggest just checking:

```c++
if (first_segment.start_time / first_segment.duration != start_number - 1) 
  return false;
```